transaction/core/src/blockchain/quorum_set.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+use alloc::{
+    string::{String, ToString},
+    vec,
+    vec::Vec,
+};
+use core::{
+    hash::{Hash, Hasher},
+    str::FromStr,
+};
+use mc_common::{NodeID, ResponderId};
+use mc_crypto_digestible::Digestible;
+use mc_crypto_keys::Ed25519Public;
+use prost::{Message, Oneof};
+use serde::{Deserialize, Serialize};
+
+/// The quorum set defining the trusted set of peers.
+#[derive(Clone, Deserialize, Digestible, Message, Ord, PartialOrd, Serialize)]
+pub struct QuorumSet {
+    /// Threshold (how many members do we need to reach quorum).
+    #[prost(uint32, required, tag = 1)]
+    pub threshold: u32,
+
+    /// Members.
+    #[prost(message, repeated, tag = 2)]
+    pub members: Vec<QuorumSetMemberWrapper>,
+}
+
+// This wrapper struct is required because of a peculiarity of prost: you cannot
+// have repeated oneof fields (like a Vec<QuorumSetMember>), so we wrap
+// QuorumSetMember in a struct which implements prost::Message. Unfortunately
+// protobuf also doesn't allow for required oneof fields, so the inner value has
+// to be optional. In practice we expect it to always be Some.
+#[derive(
+    Clone, Deserialize, Digestible, Eq, Hash, Message, Ord, PartialEq, PartialOrd, Serialize,
+)]
+pub struct QuorumSetMemberWrapper {
+    #[prost(oneof = "QuorumSetMember", tags = "1,2")]
+    pub member: Option<QuorumSetMember>,
+}
+
+/// A member in a [QuorumSet]. Can be either a [QuorumNode] or another
+/// [QuorumSet].
+#[derive(
+    Clone, Deserialize, Digestible, Eq, Hash, Oneof, Ord, PartialEq, PartialOrd, Serialize,
+)]
+#[serde(tag = "type", content = "args")]
+pub enum QuorumSetMember {
+    /// A single trusted entity with an identity.
+    #[prost(message, tag = 1)]
+    Node(QuorumNode),
+
+    /// A quorum set can also be a member of a quorum set.
+    #[prost(message, tag = 2)]
+    InnerSet(QuorumSet),
+}
```

Comment:
@eranrund - any opinions about the duping of QuorumSet here?

Reply:
Unavoidable, since the QuorumSet in the scp crate is generic over the node id and can't be used in prost/protobufs :/